JASP-Engine/JASP/R/regressionloglinear.R

```diff
@@ -15,559 +15,457 @@
-						logregression.result[[ len.logreg ]]$"Standard Error" <- as.numeric(loglm.estimates[i,2])
-
-						if (options$regressionCoefficientsConfidenceIntervals == TRUE){
-							logregression.result[[ len.logreg ]]$"Lower" <- as.numeric(lower[i])
-							logregression.result[[ len.logreg ]]$"Upper" <- as.numeric(upper[i])
-						}
-
-						logregression.result[[ len.logreg ]]$"Z" <- as.numeric(loglm.estimates[i,3])
-						logregression.result[[ len.logreg ]]$"p" <- as.numeric(loglm.estimates[i,4])
-						if (options$VovkSellkeMPR){
-						  logregression.result[[ len.logreg ]]$"VovkSellkeMPR" <- .VovkSellkeMPR(logregression.result[[ len.logreg ]]$"p")
-						}
-						len.logreg <- len.logreg + 1
-					}
-				}
-
-			} else {
-
-				len.logreg <- length(logregression.result) + 1
-				logregression.result[[ len.logreg ]] <- dotted.line
-				#logregression.result[[ len.logreg ]]$"Model" <- as.integer(m)
-
-				if (length(loglm.model$variables) > 0) {
-
-					variables.in.model <- loglm.model$variables
-
-					len.logreg <- len.logreg + 1
-
-					for (var in 1:length(variables.in.model)) {
-
-						logregression.result[[ len.logreg ]] <- dotted.line
-
-						if (base::grepl(":", variables.in.model[var])) {
-
-							# if interaction term
-
-							vars <- unlist(strsplit(variables.in.model[var], split = ":"))
-							name <- paste0(vars, collapse="\u2009\u273b\u2009")
-
-						} else {
-
-							name <- as.character(variables.in.model[ var])
-						}
-
-						logregression.result[[ len.logreg ]]$"Name" <- name
-						len.logreg <- len.logreg + 1
-					}
-				}
-			}
-
-		} else {
-
-			len.logreg <- length(logregression.result) + 1
-
-			if (length(loglm.model$variables) > 0) {
-
-				variables.in.model <- loglm.model$variables
-
-			}
-
-			len.logreg <- length(logregression.result) + 1
-			logregression.result[[ len.logreg ]] <- dotted.line
-			logregression.result[[ len.logreg ]]$"Model" <- 1
-
-		}
-
-		if (length(list.of.errors) > 1){
-			loglm.fit <- try( stats::glm( model.formula, family = poisson(), data = dataset), silent = TRUE)
-
-			if (inherits(loglm.fit, "try-error")) {
-				error <- .extractErrorMessage (loglm.fit)}
-				logregression[["error"]] <- list(errorType="badData",errorMessage = error)
-
-		} else if (length(list.of.errors) == 1){
-
-		logregression[["error"]] <- list(errorType = "badData", errorMessage = list.of.errors[[ 1 ]])
-	}
-
-	logregression[["data"]] <- logregression.result
-
-	results[["logregression"]] <- logregression
-	}
-
-	#######################################################################
-
-
-	logregressionanova[["data"]] <- logregressionanova.result
-	results[["logregressionanova"]] <- logregressionanova
-
-
-	if (perform == "init") {
+RegressionLogLinear <- function(jaspResults, dataset , options, ...) {
+  
+  # Read dataset
+  dataset <- .regLogLinReadData(dataset, options)
+  
+  ready <- length(options$factors) != 0
+  
+  # Error checking
+  .regLogLinCheckErrors(dataset, options, ready)
```

Comment:
Let's lift this `ready` out of `.regLogLinCheckErrors()` to be inline with other analyses